I15_1 (9.3pre) Rewrite MockNeXusSlit to return an NXslit rather than …

Mark-Booth opened the pull request:
…NXpositioner

Mark-Booth committed:
I15_1 (9.3pre) Rewrite MockNeXusSlit to return an NXslit rather than NXpositioner

gerring committed:
Merge branch 'master' into MockNeXusSlit-rewrite

gerring commented on org.eclipse.scanning.example/src/org/eclipse/scanning/example/scannable/MockNeXusSlit.java:
```diff
@@ -11,92 +11,177 @@
 		super();
 	}
 
+	public MockNeXusSlit(String name, double d, int level) {
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Move this constructor to comply with Java Code Conventions. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1213)

gerring commented on org.eclipse.scanning.example/src/org/eclipse/scanning/example/scannable/MockNeXusSlit.java:
```diff
@@ -11,92 +11,177 @@
+	public MockNeXusSlit(String name, double d, int level) {
+		super(name, d, level);
+	}
 	public MockNeXusSlit(String name, double d, int level, String unit) {
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Move this constructor to comply with Java Code Conventions. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1213)

gerring commented on org.eclipse.scanning.example/src/org/eclipse/scanning/example/scannable/MockNeXusSlit.java:
```diff
@@ -11,92 +11,177 @@
+		}
+	}
+
+	private void write(Number demand, Number actual, IPosition loc) throws Exception {
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this unused private "write" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00112)

Mark-Booth replied:
Erm, write is used in the public setPosition() method!

gerring commented on org.eclipse.scanning.example/src/org/eclipse/scanning/example/scannable/MockNeXusSlit.java:
```diff
@@ -11,92 +11,177 @@
+			yLzValue = positioner.initializeLazyDataset(NXslit.NX_Y_GAP, info.getRank(), Double.class);
+			xLzValue.setFillValue(fill);
+			yLzValue.setFillValue(fill);
+			xLzValue.setChunking(info.createChunk(false, 8)); // TODO Might be slow, need to check this
```
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1135)

gerring commented on org.eclipse.scanning.example/src/org/eclipse/scanning/example/scannable/MockNeXusSlit.java:
```diff
@@ -11,92 +11,177 @@
+
+	private void write(Number demand, Number actual, IPosition loc) throws Exception {
+
+		if (xLzValue==null || yLzValue==null) return;
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') At most one statement is allowed per line, but 2 statements were found on this line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00122)

gerring commented on org.eclipse.scanning.example/src/org/eclipse/scanning/example/scannable/MockNeXusSlit.java:
```diff
@@ -11,92 +11,177 @@
+			xLzValue.setFillValue(fill);
+			yLzValue.setFillValue(fill);
+			xLzValue.setChunking(info.createChunk(false, 8)); // TODO Might be slow, need to check this
+			yLzValue.setChunking(info.createChunk(false, 8)); // TODO Might be slow, need to check this
```
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1135)

gerring commented on org.eclipse.scanning.example/src/org/eclipse/scanning/example/scannable/MockNeXusSlit.java:
```diff
@@ -11,92 +11,177 @@
+	 * @param container
+	 * @throws NexusException if the attributes could not be added for any reason 
+	 */
+	private static void registerAttributes(NXobject nexusObject, IScanAttributeContainer container) throws NexusException {
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this unused private "registerAttributes" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)

Mark-Booth replied:
This is used in the public getNexusProvider() method!

gerring commented on org.eclipse.scanning.example/src/org/eclipse/scanning/example/scannable/MockNeXusSlit.java:
```diff
@@ -11,92 +11,177 @@
+	private static void registerAttributes(NXobject nexusObject, IScanAttributeContainer container) throws NexusException {
+		// We create the attributes, if any
+		nexusObject.setField("name", container.getName());
+		if (container.getScanAttributeNames()!=null) for(String attrName : container.getScanAttributeNames()) {
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') At most one statement is allowed per line, but 2 statements were found on this line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00122)

gerring commented on org.eclipse.scanning.example/src/org/eclipse/scanning/example/scannable/MockNeXusSlit.java:
```diff
@@ -11,92 +11,177 @@
+		final NXslit positioner = NexusNodeFactory.createNXslit();
+
+		if (info.getScanRole(getName()) == ScanRole.METADATA) {
+			//positioner.setField(NXslit.NX_X_GAP, getPosition().doubleValue());
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

gerring commented on org.eclipse.scanning.example/src/org/eclipse/scanning/example/scannable/MockNeXusSlit.java:
```diff
@@ -11,92 +11,177 @@
+		if (container.getScanAttributeNames()!=null) for(String attrName : container.getScanAttributeNames()) {
+			try {
+				nexusObject.setField(attrName, container.getScanAttribute(attrName));
+			} catch (Exception e) {
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1166)

Mark-Booth replied:
SonarQube appears to be missing that we now throw without losing the exception, just as in the example.

gerring commented on org.eclipse.scanning.example/src/org/eclipse/scanning/example/scannable/MockNeXusSlit.java:
```diff
@@ -11,92 +11,177 @@
+
+	private ILazyWriteableDataset xLzSet;
+	private ILazyWriteableDataset yLzSet;
+	private ILazyWriteableDataset xLzValue;
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Move this variable to comply with Java Code Conventions. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1213)

gerring commented on org.eclipse.scanning.example/src/org/eclipse/scanning/example/scannable/MockNeXusSlit.java:
```diff
@@ -11,92 +11,177 @@
+	private ILazyWriteableDataset xLzSet;
+	private ILazyWriteableDataset yLzSet;
+	private ILazyWriteableDataset xLzValue;
+	private ILazyWriteableDataset yLzValue;
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Move this variable to comply with Java Code Conventions. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1213)

gerring commented on org.eclipse.scanning.example/src/org/eclipse/scanning/example/scannable/MockNeXusSlit.java:
```diff
@@ -11,92 +11,177 @@
+			positioner.setY_gapScalar(getPosition().doubleValue());
+		} else {
+			String floatFill = System.getProperty("GDA/gda.nexus.floatfillvalue", "nan");
+			double fill = floatFill.equalsIgnoreCase("nan") ? Double.NaN : Double.parseDouble(floatFill);
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Move the "nan" string literal on the left side of this string comparison. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1132)

gerring commented on org.eclipse.scanning.example/src/org/eclipse/scanning/example/scannable/MockNeXusSlit.java:
```diff
@@ -11,92 +11,177 @@
+		if (demand!=null) {
+			int index = loc.getIndex(getName());
+			if (index<0) {
+				throw new Exception("Incorrect data index for scan for value of '"+getName()+"'. The index is "+index);
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00112)

gerring commented on org.eclipse.scanning.example/src/org/eclipse/scanning/example/scannable/MockNeXusSlit.java:
```diff
@@ -11,92 +11,177 @@
+	private ILazyWriteableDataset xLzValue;
+	private ILazyWriteableDataset yLzValue;
+
+	private boolean writingOn = true;
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Move this variable to comply with Java Code Conventions. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1213)

gerring commented on org.eclipse.scanning.example/src/org/eclipse/scanning/example/scannable/MockNeXusSlit.java:
```diff
@@ -11,92 +11,177 @@
+
+	private boolean writingOn = true;
 
 	public MockNeXusSlit() {
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Move this constructor to comply with Java Code Conventions. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1213)

gerring commented on org.eclipse.scanning.example/src/org/eclipse/scanning/example/scannable/MockNeXusSlit.java:
```diff
@@ -11,92 +11,177 @@
+		if (value!=null) {
+			int index = position!=null ? position.getIndex(getName()) : -1;
+			if (isRealisticMove()) {
+				value = doRealisticMove(value, index, -1);
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Introduce a new variable instead of reusing the parameter "value". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1226)

gerring commented:
SonarQube analysis reported 21 issues
* ![MAJOR][MAJOR] 9 major
* ![MINOR][MINOR] 10 minor
* ![INFO][INFO] 2 info

Watch the comments in this conversation to review them.

[INFO]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO'
[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

Mark-Booth committed:
SonarQube review

gerring commented on org.eclipse.scanning.example/src/org/eclipse/scanning/example/scannable/MockNeXusSlit.java:
```diff
@@ -11,92 +11,180 @@
 		super(name, d, level, unit);
 	}
+	
+	public boolean isWritingOn() {
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Move this constructor to comply with Java Code Conventions. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1213)

gerring commented on org.eclipse.scanning.example/src/org/eclipse/scanning/example/scannable/MockNeXusSlit.java:
```diff
@@ -11,92 +11,180 @@
+	public boolean isWritingOn() {
+		return writingOn;
+	}
 
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Move this constructor to comply with Java Code Conventions. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1213)

gerring commented on org.eclipse.scanning.example/src/org/eclipse/scanning/example/scannable/MockNeXusSlit.java:
```diff
@@ -11,92 +11,180 @@
+	}
+
+	private void write(Number demand, Number actual, IPosition loc) throws Exception {
+
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this unused private "write" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00112)

gerring commented on org.eclipse.scanning.example/src/org/eclipse/scanning/example/scannable/MockNeXusSlit.java:
```diff
@@ -11,92 +11,180 @@
+			xLzValue.setFillValue(fill);
+			yLzValue.setFillValue(fill);
+			xLzValue.setChunking(info.createChunk(false, 8)); // Might be slow, need to check this
+			yLzValue.setChunking(info.createChunk(false, 8)); // Might be slow, need to check this
```
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1135)

gerring commented on org.eclipse.scanning.example/src/org/eclipse/scanning/example/scannable/MockNeXusSlit.java:
```diff
@@ -11,92 +11,180 @@
+	private void write(Number demand, Number actual, IPosition loc) throws Exception {
+
+		if (xLzValue==null || yLzValue==null) {
+			return;
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') At most one statement is allowed per line, but 2 statements were found on this line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00122)

gerring commented on org.eclipse.scanning.example/src/org/eclipse/scanning/example/scannable/MockNeXusSlit.java:
```diff
@@ -11,92 +11,180 @@
+			yLzValue.setFillValue(fill);
+			xLzValue.setChunking(info.createChunk(false, 8)); // Might be slow, need to check this
+			yLzValue.setChunking(info.createChunk(false, 8)); // Might be slow, need to check this
+			xLzValue.setWritingAsync(true);
```
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1135)

gerring commented on org.eclipse.scanning.example/src/org/eclipse/scanning/example/scannable/MockNeXusSlit.java:
```diff
@@ -11,92 +11,180 @@
+	/**
+	 * Add the attributes for the given attribute container into the given nexus object.
+	 * @param positioner
+	 * @param container
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this unused private "registerAttributes" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)

gerring commented on org.eclipse.scanning.example/src/org/eclipse/scanning/example/scannable/MockNeXusSlit.java:
```diff
@@ -11,92 +11,180 @@
+	public NexusObjectProvider<NXslit> getNexusProvider(NexusScanInfo info) throws NexusException {
+		final NXslit positioner = NexusNodeFactory.createNXslit();
+
+		if (info.getScanRole(getName()) == ScanRole.METADATA) {
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

gerring commented on org.eclipse.scanning.example/src/org/eclipse/scanning/example/scannable/MockNeXusSlit.java:
```diff
@@ -11,92 +11,180 @@
+	private boolean writingOn = true;
+	
 	public MockNeXusSlit() {
 		super();
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Move this variable to comply with Java Code Conventions. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1213)

gerring commented on org.eclipse.scanning.example/src/org/eclipse/scanning/example/scannable/MockNeXusSlit.java:
```diff
@@ -11,92 +11,180 @@
+				yLzValue.setSlice(null, newActualPositionData, ySliceND);
+			}
+		}
+
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') At most one statement is allowed per line, but 2 statements were found on this line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00122)

gerring commented on org.eclipse.scanning.example/src/org/eclipse/scanning/example/scannable/MockNeXusSlit.java:
```diff
@@ -11,92 +11,180 @@
+	
 	public MockNeXusSlit() {
 		super();
 	}
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Move this variable to comply with Java Code Conventions. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1213)

gerring commented on org.eclipse.scanning.example/src/org/eclipse/scanning/example/scannable/MockNeXusSlit.java:
```diff
@@ -11,92 +11,180 @@
 	public MockNeXusSlit() {
 		super();
 	}
+	
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Move this variable to comply with Java Code Conventions. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1213)

gerring commented on org.eclipse.scanning.example/src/org/eclipse/scanning/example/scannable/MockNeXusSlit.java:
```diff
@@ -11,92 +11,180 @@
 		super();
 	}
+	
+	public MockNeXusSlit(String name, double d, int level) {
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Move this variable to comply with Java Code Conventions. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1213)

gerring commented on org.eclipse.scanning.example/src/org/eclipse/scanning/example/scannable/MockNeXusSlit.java:
```diff
@@ -11,92 +11,180 @@
+		} else {
+			String floatFill = System.getProperty("GDA/gda.nexus.floatfillvalue", "NaN");
+			double fill = "NaN".equalsIgnoreCase(floatFill) ? Double.NaN : Double.parseDouble(floatFill);
+
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Move the "nan" string literal on the left side of this string comparison. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1132)

gerring commented on org.eclipse.scanning.example/src/org/eclipse/scanning/example/scannable/MockNeXusSlit.java:
```diff
@@ -11,92 +11,180 @@
+		if (xLzSet==null || yLzSet==null) {
+			return;
+		}
+		if (demand!=null) {
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00112)

gerring commented on org.eclipse.scanning.example/src/org/eclipse/scanning/example/scannable/MockNeXusSlit.java:
```diff
@@ -11,92 +11,180 @@
+	
+	public MockNeXusSlit(String name, double d, int level) {
+		super(name, d, level);
+	}
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Move this variable to comply with Java Code Conventions. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1213)

gerring commented on org.eclipse.scanning.example/src/org/eclipse/scanning/example/scannable/MockNeXusSlit.java:
```diff
@@ -11,92 +11,180 @@
+		super(name, d, level);
+	}
 
 	public MockNeXusSlit(String name, double d, int level, String unit) {
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Move this constructor to comply with Java Code Conventions. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1213)

gerring commented on org.eclipse.scanning.example/src/org/eclipse/scanning/example/scannable/MockNeXusSlit.java:
```diff
@@ -11,92 +11,180 @@
+			int index = position!=null ? position.getIndex(getName()) : -1;
+			if (isRealisticMove()) {
+				value = doRealisticMove(value, index, -1);
+			}
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Introduce a new variable instead of reusing the parameter "value". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1226)

gerring commented on org.eclipse.scanning.example/src/org/eclipse/scanning/example/scannable/MockNeXusSlit.java:
```diff
@@ -11,92 +11,180 @@
+		}
+	}
+
+	private void write(Number demand, Number actual, IPosition loc) throws Exception {
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this unused private "write" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00112)

Mark-Booth replied:
write is used in the public setPosition() method!

Mark-Booth replied:
Suggestions for which exception to throw appreciated...

gerring commented on org.eclipse.scanning.example/src/org/eclipse/scanning/example/scannable/MockNeXusSlit.java:
```diff
@@ -11,92 +11,180 @@
+		if (value!=null) {
+			int index = position!=null ? position.getIndex(getName()) : -1;
+			if (isRealisticMove()) {
+				value = doRealisticMove(value, index, -1);
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Introduce a new variable instead of reusing the parameter "value". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1226)

Mark-Booth replied:
SonarQube doesn't seem to have noticed that the latest commit fixes this!

gerring commented on org.eclipse.scanning.example/src/org/eclipse/scanning/example/scannable/MockNeXusSlit.java:
```diff
@@ -11,92 +11,180 @@
+		if (demand!=null) {
+			int index = loc.getIndex(getName());
+			if (index<0) {
+				throw new Exception("Incorrect data index for scan for value of '"+getName()+"'. The index is "+index);
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00112)

Mark-Booth replied:
Suggestions for which exception to throw appreciated...

gerring commented:
SonarQube analysis reported 6 issues
* ![MAJOR][MAJOR] 5 major
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

Mark-Booth committed:
SonarQube re-review

gerring commented:
SonarQube analysis reported 4 issues
* ![MAJOR][MAJOR] 4 major

Watch the comments in this conversation to review them.

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

Mark-Booth committed:
I15_1 (9.3pre) Disable testScanWithConfiguredScannable

The scannable under test is dcs, which is supposed to be an NXslit, but
this test assumes it will be an NXpositioner.
